src/cpp/core/Base64.cpp

```diff
@@ -70,45 +140,187 @@ Error encode(const FilePath& inputFile, std::string* pOutput)
    return encode(contents, pOutput);
 }
 
+namespace {
 
-Error decode(const std::string& input, std::string* pOutput)
+std::size_t decoded_size(std::size_t n)
 {
-   using namespace boost::archive::iterators;
+   return n * 3 / 4;
+}
+
+class Decoder
```

Comment:
are instances of this class copyable and/or reusable? (let's document either way)